src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -32,12 +32,11 @@
 
 import static com.github.kagkarlsson.scheduler.ExecutorUtils.defaultThreadFactoryWithPrefix;
 
-public class Scheduler implements SchedulerClient {
+public class Scheduler extends SchedulerClient.StandardSchedulerClient {
 
```

Comment:
I think I would prefer keeping the interface and delegate to internal `StandardSchedulerClient`